### PR TITLE
Implement paginated search function

### LINKED ISSUE #7: Only first page of search results are accessible
When calling the `OntoPortalClient.search` method the response indicates that there are more results:

```python
>>> client = BioPortalClient()
>>> response = client.search("swim")
>>> response["totalCount"]
575
>>> len(response["collection"])
50
```

There doesn't appear to be any way to access all 575 results in the above example.

### PR DESCRIPTION
**Pull request by pkalita-lbl**
Closes #7

### CODE REVIEW

**Comment by cthoyt** [comment]
@pkalita-lbl i like the idea in theory - can you please look into the tests and fix the code style before I carefully review it? The ci feedback is below

**Comment by pkalita-lbl** [comment]
Absolutely. Taking a look now.

File: .gitignore [hunk 1]
```diff
@@ -463,11 +463,6 @@ tags
 
 ### VisualStudioCode ###
 .vscode/*
-!.vscode/settings.json
```
Comment (cthoyt) [inline .gitignore:466]: why are these removed?

File: src/ontoportal_client/api.py [hunk 2]
```diff
@@ -110,7 +110,13 @@ def search(self, text: str, ontology: Optional[str] = None):
         params = {"q": text, "include": ["prefLabel"]}
         if ontology:
             params["ontologies"] = ontology
-        return self.get_json("/search", params)
+        page = 1
+        while page:
```
Comment (cthoyt) [inline src/ontoportal_client/api.py]: this is a good place for the walrus operator

File: .gitignore (hunk 1 above)
Comment (pkalita-lbl) [reply .gitignore:466]: I guess I generally haven't found checking in anything under `.vscode` to be useful, especially in projects where developers aren't all necessarily using the same IDE. But I don't really have strong feelings on it, and I'm happy to revert this and check in my `settings.json` instead.

**Comment by pkalita-lbl** [comment]
It looks like all the test failures are due to missing API keys (I have to admit I don't even have personal keys for a lot of these services). I assume these tests work in the main repo because there are keys stored as repo secrets?

File: src/ontoportal_client/api.py (hunk 2 above)
Comment (pkalita-lbl) [reply src/ontoportal_client/api.py]: Oh yeah, I forgot that exists! I gave it a try but what I ended up with felt very convoluted in order to only get the default `1` on the first iteration. Or maybe I'm not quite grasping what you had in mind.

**Comment by pkalita-lbl** [comment]
Sorry, my last comment was worded in a confusing way and I should have done a little more research first anyway. 

So, yeah, I can see that the main repo has secrets for API keys that get [passed to the test environment](https://github.com/cthoyt/ontoportal-client/blob/main/.github/workflows/tests.yml#L73-L78). However in this case, those secrets aren't being passed because the [workflow is being triggered from a forked repo](https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow).

@cthoyt What do you think is the right thing to do here? Try and move these changes to a branch of the main repo and open a new PR from there?

File: src/ontoportal_client/api.py [hunk 3]
```diff
@@ -110,7 +110,12 @@ def search(self, text: str, ontology: Optional[str] = None):
         params = {"q": text, "include": ["prefLabel"]}
         if ontology:
             params["ontologies"] = ontology
-        return self.get_json("/search", params)
+        page = "1"
+        while page:
+            params["page"] = page
+            result = self.get_json("/search", params)
+            yield result
+            page = result["nextPage"]
```
Comment (cthoyt) [inline src/ontoportal_client/api.py]: this seems to work but the test doesn't represent what happens on the last page. Is there a `None` for `result["nextPage"]`? or does it blow up with a `KeyError` because that's not present at all? we need a test for this

Comment (pkalita-lbl) [reply src/ontoportal_client/api.py]: Yes, `result["nextPage"]` will be `None` on the last page. Do you think it would be better to be more defensive and use `result.get("nextPage", None)` just in case?

The test I added called [`test_search_pagination`](https://github.com/pkalita-lbl/ontoportal-client/blob/a855f53f822db30443508ab61f21a36be70c00fe/tests/test_client.py#L32) that was intended to verify that the generator can be exhausted (and therefore the last page reached) without anything bad happening. Did you have something else in mind?

Comment (cthoyt) [reply src/ontoportal_client/api.py]: Alright thanks for clarifying, I simplified the implementation a bit and added some docs. I'm pretty confident in this now.

**Comment by cthoyt** [comment]
@pkalita-lbl I added you as a contributor so you should be able to make your own branches for next time. For now, I'm going to just wing it and merge this since tox is working locally.

File: .gitignore (hunk 1 above)
Comment (cthoyt) [reply .gitignore:466]: Oh okay, it's definitely the case that I don't want you committing IDE settings. Weird that this is even in here, does this imply that the gitignore.io wants people committing this?

Comment (pkalita-lbl) [reply .gitignore:466]: I haven't done an immense amount of research on it, but I think it's one of those things where reasonable people can have differing opinions. Like, my `settings.json` has stuff in it that says "this project uses `pytest` instead of `unittest`" which could be beneficial for everyone, but it _could_ also have stuff in it that's like "I want my editor to have a larger font for this project". So I err on the conservative side and leave it out generally.

File: src/ontoportal_client/api.py (hunk 3 above)
Comment (pkalita-lbl) [reply src/ontoportal_client/api.py]: Nice, I like your changes. Thanks!

File: .gitignore (hunk 1 above)
Comment (cthoyt) [reply .gitignore:466]: yeah, for sure this stuff doesn't belong in version control
